src/v/http/include/http/client.h

```diff
@@ -287,13 +287,20 @@ auto with_client(client&& cl, Func func) {
       std::is_nothrow_move_constructible_v<Func>,
       "Func's move constructor must not throw");
     return ss::do_with(
-      std::move(cl), [func = std::move(func)](client& cl) mutable {
-          return ss::futurize_invoke(func, cl).finally([&cl] {
-              return cl.stop().then([&cl] {
-                  cl.shutdown();
-                  return ss::make_ready_future<>();
-              });
-          });
+      false, std::move(cl), [func = std::move(func)](bool stopped, client& cl) mutable {
+          return ss::futurize_invoke(func, cl)
+            .handle_exception_type([&stopped](ss::tls::verification_error err) {
```

Comment:
@abhijat is this sufficent? I'm thinking there may be other cases where this exception type isn't handeled in `client.cc` and therefore with this patch its possible calls to stop() may be skipped.

Another solution i had was to set a flag when client::stop() is called, then exposing a getter for it, checking that in the finally block

Reply:
I like the idea of keeping state in the client, we do that[ in a few places](https://github.com/redpanda-data/redpanda/blob/dev/src/v/cloud_storage/remote_segment.cc#L242-L245) with a boolean `_stopped` field or something similar, it could even be checked within `client::stop` and then do an early return if `_stopped` is already set.